src/EventStore.Projections.Core/Services/Interpreted/JintProjectionStateHandler.cs

```diff
@@ -993,9 +987,266 @@ private bool EnsureLinkMetadata(out JsValue value)
 
 			public void HandleDeleted(JsValue state, string partition, bool isSoftDelete) {
 				if (_deleted != null) {
-					_deleted.Call(this, new JsValue[] {state, Null, partition, isSoftDelete });
+					_deleted.Call(this, new JsValue[] { state, Null, partition, isSoftDelete });
+				}
+			}
+
+			private readonly Serializer _serializer = new Serializer();
+			public string Serialize(JsValue value) {
+				return JintProjectionStateHandler.Serialize(_serializer, value);
+			}
+		}
+
+		public static string Serialize(JsValue value) {
+			return Serialize(new Serializer(), value);
+		}
+
+		private static string Serialize(Serializer serializer, JsValue instance) {
+			var serialized = serializer.Serialize(instance);
+			return Encoding.UTF8.GetString(serialized.Span);
+		}
+
+		private class Serializer {
+			private readonly IWriter[] _iterators;
+			private readonly ArrayBufferWriter<byte> _bufferWriter;
+			private Utf8JsonWriter _writer;
+			private Dictionary<string, JsonEncodedText> _knownPropertyNames;
+			private int _depth;
+			private static readonly IWriter _completed = new CompletedWriter();
+			public Serializer() {
+				_iterators = new IWriter[64];
```

Comment:
was there a max depth of 64 (implicit) in the previous implementation?

Reply:
no